projects/Douglas Mentoring/day1.py

```diff
@@ -0,0 +1,92 @@
+import getpass
+import netmiko
+# json was here bc as I'm using textfsm to pull data from cisco devices
+# it comes in json i needed to show the output before deciding how to tread the date
+import json
+import csv
+
+# Where i'm pulling the devices list / Soon I'll make this automated which will pull direct from NARF
+file = open(r'/home/segattod/segattod-workspace/src/Segattod-package/myproject/Light_level_checks/list.txt',
+            'r')
+file = file.read().splitlines()
+
+username = input('User Name:')
+password = getpass.getpass()
+
+# Variable for "append" the data later
+formatted_data = []
+
+# for now just to name the file / Soon to add for user to add input :)
+whid = 'YEG1'
+
+
+# Function to sorte the data
+def light_level_check(connection):
+    # Command to checks light level per interface
+    transceiver_output = connection.send_command('show interface transceiver', read_timeout=90,
+                                                 use_textfsm='~/ntc-templates/ntc_templates/templates/cisco_ios_show_interface_transceiver.textfsm')
+
+    for x in transceiver_output:
+        # added both variables below to do a check at that moment based on the "output" for that device interface the loop is going through
+        # why? bc was the way i could think to append(look below) all data to export later in CSV
+        # basically i use the "entry" information from "x" do all the checks and move to the next, and so on.
+        sfpmodel = connection.send_command(f'sh idprom interface {x["iface"]} detail | i Vendor Name')
+        sfpspeed = connection.send_command(
+            f'sh idprom interface {x["iface"]} detail | i Administrative Speed')
```

Comment:
There is a design Principle called DRY: "Don't Repeat Yourself".  If you re typing the same thing more than once, it's likely faster and easier to use a variable.   Try this
```suggestion
        base_command = f'sh idprom interface {x["iface"]} detail | i "
        sfpmodel = connection.send_command(f"{base_command}Vendor Name')
        sfpspeed = connection.send_command(f"{base_command}Administrative Speed')
```


Similarly, sftmodel and sftspeed are not very "pythonic" variable names.   At a miniumumm, I'd suggest `sfp_model` and `sfp_speed` but really, that isn't what these are and you've painted yourself in a corner below trying to think of new names.  Perhaps try `model_data` and `speed_data` or even `raw_model_info` and `raw_speed_info`.

Don't be afraid of long names, they make life easier when you come back to this in a month and forget what everything meant.